modules/overrides/overrides_tenants_http.go

```diff
@@ -0,0 +1,83 @@
+package overrides
+
+import (
+	_ "embed" // Used to embed html templates
+	"html/template"
+	"net/http"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/grafana/tempo/pkg/util"
+)
+
+//go:embed tenants.gohtml
+var tenantsPageHTML string
+var tenantsTemplate = template.Must(template.New("webpage").Parse(tenantsPageHTML))
+
+type tenantsPageContents struct {
+	Now     time.Time           `json:"now"`
+	Tenants []tenantsPageTenant `json:"tenants,omitempty"`
+}
+
+type tenantsPageTenant struct {
+	Name                         string `json:"name"`
+	HasRuntimeOverrides          bool   `json:"has_runtime_overrides"`
+	HasUserConfigurableOverrides bool   `json:"has_user_configurable_overrides"`
+}
+
+func TenantsHandler(o Interface) http.HandlerFunc {
+	return func(w http.ResponseWriter, req *http.Request) {
+		tenants := make(map[string]tenantsPageTenant)
+
+		// runtime overrides
+		var runtimeTenants []string
+		switch o := o.(type) {
+		case *runtimeConfigOverridesManager:
+			runtimeTenants = o.GetTenantIDs()
+		case *userConfigurableOverridesManager:
+			runtimeTenants = o.Interface.GetTenantIDs()
+		default:
+			util.WriteTextResponse(w, "Internal error happened when retrieving runtime overrides")
+		}
+		for _, tenant := range runtimeTenants {
+			tenants[tenant] = tenantsPageTenant{
+				Name:                         tenant,
+				HasRuntimeOverrides:          true,
+				HasUserConfigurableOverrides: false,
+			}
+		}
+
+		// user-configurable overrides
+		userConfigurableOverridesManager, ok := o.(*userConfigurableOverridesManager)
+		if ok {
+			for _, tenant := range userConfigurableOverridesManager.GetTenantIDs() {
+				_, hasRuntimeOverrides := tenants[tenant]
+
+				tenants[tenant] = tenantsPageTenant{
```

Comment:
nit: Can this be simplified?  Either by using `*tenantsPageTenant` or making use of how `tenants[tenant]` will return an uninitialized struct (with HasRunTimeOverrides=false)